# Assignment 6

To measure the psychological state of Americans, a psychology researcher (in 1929) devised
three questions put to 3443 Americans from different walks of life. He asked them whether they
were happy. They could answer only **yes** or **no** to each question. By today’s standards the study
is quite simple, and he did no formal statistical analysis of his data. The frequency data (number
of **yes** and **no** responses within each employment category) from this study are in the file
*Americandata.csv*.

---

You need to use the data and answer the question, that is, **Is the type of profession related to the happiness of American people?**
Report the **χ2** statistic and the *statistical significance of test*, in addition to the contingency table.
Also, explain the results and what they mean. What does the value of **χ2** statistic mean? Is it
good or bad? Is the test statistically significant?

In [2]:
import pandas as pd
from scipy.stats import chi2_contingency

df = pd.read_csv("Americandata.csv", index_col=0)

# Contingency table with totals
table = df.copy()
table["Total"] = table.sum(axis=1)
table.loc["Total"] = table.sum()
print("Observed contingency table:")
display(table)

chi2, p, dof, expected = chi2_contingency(df)
expected_df = pd.DataFrame(expected, index=df.index, columns=df.columns).round(2)
print("Expected counts:")
display(expected_df)

print(f"Chi-square = {chi2:.2f}")
print(f"Degrees of freedom = {dof}")
print(f"p-value = {p:.3e}")
print(f"Smallest expected count = {expected.min():.2f}")

Observed contingency table:


,Students,Laborers,Preachers,Physicians,Housewives,Teachers,Lawyers,Musicians,Total
Happy,,,,,,,,,
Yes,390,378,35,159,78,108,11,31,1190
No,1610,122,265,51,122,38,64,19,2291
Total,2000,500,300,210,200,146,75,50,3481


Expected counts:


,Students,Laborers,Preachers,Physicians,Housewives,Teachers,Lawyers,Musicians
Happy,,,,,,,,
Yes,683.71,170.93,102.56,71.79,68.37,49.91,25.64,17.09
No,1316.29,329.07,197.44,138.21,131.63,96.09,49.36,32.91


Chi-square = 936.14
Degrees of freedom = 7
p-value = 7.523e-198
Smallest expected count = 17.09


Put your analysis in the empty `markdown` cell below.

The Chi-Square test
    x^2 = 936.14 with 7 degrees of freedom
    p = 7.5 x 10^-198, which is pretty much 0

All of the expected counts are above 5, meeting the reqs for the test

Since the p value is practically 0, we're going to reject the null hypothesis